### PR TITLE
add amp runtime api

### PR DESCRIPTION
**Pull request by sebastianbenz**

### CODE REVIEW

File: runtime-version/lib/MaxAge.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  }
+
+  /**
+   * Parses a max-age header string and returns a new MaxAge instance.
+   *
+   * @param {string} str a max-age header string
+   * @returns {MaxAge}
+   */
+  static parse(string) {
+    const match = string.match(/max-age=([0-9]+)[^0-9]?/i);
```
Comment (andreban) [inline runtime-version/lib/MaxAge.js]: Can we add a check for null to string? eg: `if (!string) return MaxAge.zero();`

File: runtime-version/lib/MaxAge.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    let maxAge = 0;
+    if (match) {
+      maxAge = parseInt(match[1], 10);
+    }
+    return new MaxAge(Date.now(), maxAge);
```
Comment (andreban) [inline runtime-version/lib/MaxAge.js]: What's effectively the difference between this and MaxAge.zero()? Could it be worth to create a MaxAge.now(), and re-use it when checking for the null string check above?

File: runtime-version/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{
+  "extends": "google",
+  "installedESLint": true,
+  // http://eslint.org/docs/rules/
```
Comment (andreban) [inline runtime-version/.eslintrc.json]: nit: Comments aren't allowed in JSON. A common solution is to use a "_comment" field to add the comment and not end up with an invalid JSON. ESLint doesn't seem to care about this though.

File: runtime-version/lib/MaxAge.js (hunk 2 above)
Comment (sebastianbenz) [reply runtime-version/lib/MaxAge.js]: made this more explicit

File: runtime-version/lib/MaxAge.js (hunk 1 above)
Comment (sebastianbenz) [reply runtime-version/lib/MaxAge.js]: done

File: runtime-version/lib/RuntimeVersion.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+   */
+  currentVersion(options = {}) {
+    if (options.canary) {
+      return this.getAndUpdateCanary_();
+    }
+    return this.getAndUpdateRelease_();
+  }
+
+  /* PRIVATE */
+  getAndUpdateRelease_() {
```
[the remark this answers is not in the document]
Comment (sebastianbenz) [reply runtime-version/lib/RuntimeVersion.js]: let's standardise on postfixing

**Comment by sebastianbenz** [comment]
PTAL

File: runtime-version/lib/RuntimeVersion.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (andreban) [reply runtime-version/lib/RuntimeVersion.js]: SGTM

**Pull request merged**